File: VB/Models/Customer.vb
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.Data.Entity

Namespace GridViewOptimisticConcurrencyMvc.Models

    Public Class Customer

        <Key>
        Public Property Id As Integer

        Public Property FirstName As String

        Public Property LastName As String

        Public Property Age As Integer

        Public Property Email As String

        <Timestamp>
        Public Property RowVersion As Byte()
    End Class

    Public Class CustomerDbContext
        Inherits DbContext

        Public Sub New()
            MyBase.New("CustomerDbContext")
            Call Database.SetInitializer(New CustomerDbContextInitializer())
        End Sub

        Public Property Customers As DbSet(Of Customer)
    End Class

    Public Class CustomerDbContextInitializer
        Inherits DropCreateDatabaseIfModelChanges(Of CustomerDbContext)

        Protected Overrides Sub Seed(ByVal context As CustomerDbContext)
            Dim defaultCustomers As IList(Of Customer) = New List(Of Customer)()
            defaultCustomers.Add(New Customer() With {.FirstName = "David", .LastName = "Adler", .Age = 25, .Email = "David.Adler@somewhere.com"})
            defaultCustomers.Add(New Customer() With {.FirstName = "Michael", .LastName = "Alcamo", .Age = 38, .Email = "Michael.Alcamo@somewhere.com"})
            defaultCustomers.Add(New Customer() With {.FirstName = "Amy", .LastName = "Altmann", .Age = 27, .Email = "Amy.Altmann@somewhere.com"})
            For Each std As Customer In defaultCustomers
                context.Customers.Add(std)
            Next

            MyBase.Seed(context)
        End Sub
    End Class
End Namespace

File: CS/Models/Customer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Data.Entity;

namespace GridViewOptimisticConcurrencyMvc.Models {
    public class Customer {
        [Key]
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public int Age { get; set; }
        public string Email { get; set; }
        [Timestamp]
        public Byte[] RowVersion { get; set; }
    }

    public class CustomerDbContext : DbContext {
        public CustomerDbContext()
            : base("CustomerDbContext") {
            Database.SetInitializer(new CustomerDbContextInitializer());
        }

        public DbSet<Customer> Customers { get; set; }
    }

    public class CustomerDbContextInitializer : DropCreateDatabaseIfModelChanges<CustomerDbContext> {
        protected override void Seed(CustomerDbContext context) {
            IList<Customer> defaultCustomers = new List<Customer>();

            defaultCustomers.Add(new Customer() { FirstName = "David", LastName = "Adler", Age = 25, Email = "David.Adler@somewhere.com" });
            defaultCustomers.Add(new Customer() { FirstName = "Michael", LastName = "Alcamo", Age = 38, Email = "Michael.Alcamo@somewhere.com" });
            defaultCustomers.Add(new Customer() { FirstName = "Amy", LastName = "Altmann", Age = 27, Email = "Amy.Altmann@somewhere.com" });

            foreach (Customer std in defaultCustomers)
                context.Customers.Add(std);

            base.Seed(context);
        }
    }
}